_posts/2021-06-28-regular-expressions-in-ruby.md

```diff
@@ -0,0 +1,146 @@
+
+Let's start by searching for strings in a way we already know.
+
+Here we check the string for the presence of a word `Nick`, the result, respectively, returns us `true`:
+
+```ruby
+'Neil and Nick'.include?('Nick')
+=> true
+```
+
+And in this case, the word `Alex` in our line is missing - we get the result `false`
+
+```ruby
+'Neil and Nick'.include?('Alex')
+=> false
+```
+
+Now, using the same example, we will use a regular expression, namely, we will use a method `match` whose argument is the desired value enclosed by slashes, i.e. `/Neil/...` The method returns the found value to us in the form of a class object `MatchData`, and more specifically the name `Neil` in encoded form:
+
+```ruby
+'Neil and Nick'.match(/Neil/)
+=> #<MatchData "\x83\xAE\xE8\xAO">
+```
+
+In the case of using a word that the string does not contain, we get nil.
+
+We can also use a more concise method `=~` that does a similar job:
+
+```ruby
+'Neil and Nick' =~ ('Alex')
+=> nil
+```
+
+Substitute the correct word and we see that the method returned to us `9`. This number means the position (character number) in the string where the match was first encountered (the required condition). Let's check if the result is correct:
+
+```ruby
+'Neil and Nick' =~ ('Nick')
+=> true
+```
+
+We can set more complex constructions, for example, we need to find all the numbers, or skip only those data that match the regular expression (often you will have to use this approach when checking email).
+
+Consider the following example:
+
+```ruby
+'cat' =~ /c.t/
+=> 0
+```
+
+A period in a regular expression means that any single character is possible in its place , be it `ONE` number, `ONE` letter, or any other character.
+
+Let us now apply the anchor in our regular expression ( anchor ) `^`- signifying the beginning of the line. When using this anchor, we explicitly indicate that the character сshould come first. In the case of a different character in the first place after the anchor (the beginning of the line), the method will return us `nil`:
+
+``` ruby
+'cat' =~ /^c.t/
+=> 0
+
+'1cat' =~ /^c.t/
+=> nil
+```
+
+And now we will introduce a restriction on the last letter, that is, the letter `t` must be the last in the line, otherwise the line will not pass our 'funnel'. To do this, we will use an anchor `$`- denoting the end of the line:
+
+```ruby
+'cat' =~ /^c.t$/
+=> 0
+
+'cute' =~ /^c.t$/
+=> nil
+```
+
+In the last example, the letter is tnot the last character, hence the result nil.
+
+It's time to get acquainted with the quantifiers ( Quantifier ) from the English word Quantify - which means to determine the amount . Let's take an example already familiar to us, and substitute an asterisk symbol *( asterisk , asterisk ) instead of a dot . Using this quantifier, we can enter any number of characters from zero to infinity in place of the asterisk:
```

Comment:
```suggestion
It's time to get acquainted with the quantifiers ( Quantifier ) from the English word Quantify - which means to determine the amount. Let's take an example already familiar to us, and substitute an asterisk symbol *(asterisk) instead of a dot. Using this quantifier, we can enter any number of characters from zero to infinity in place of the asterisk:
```